### DISCUSSION TITLE
Using Camomile on an Existing Pure Data Patch

### DISCUSSION THREAD

**Discussion by Metronomitity** [pierreguillot/Camomile#268] [General]
Hello, I have some very basic questions about Camomile.

Following the documentation, I have successfully loaded AlmondOrgan into my DAW (Cakewalk by Bandlab). I copied Camomile.vst3 from the Camomile folder to the AlmondOrgan folder, changed the file's name to AlmondOrgan.vst3, and copied the AlmondOrgan folder from the Examples folder to C:/Users/Program Files/Common Files/VST3, and Cakewalk then found and loaded the patch. I also got MiniMock to work using the same process.

However, I cannot get a Pure Data patch other than the patches in the Examples folder to work. I have been following a YouTube series showing the development of a Pure Data synthesizer, and wanted to load that synthesizer into Cakewalk. I downloaded and unzipped the zip file (it is the final version, lesson_16.zip, on this website: http://lwmmusic.com/learning-lswpd-1.html), copied Camomile.vst3 from the Camomile folder to the lwm_fam2_synthesizer folder (which is what's inside the lesson_16 download), changed the file's name to lwm_fam2_synthesizer.vst3, and copied the lwm_fam2_synthesizer folder to C:/Users/Program Files/Common Files/VST3. When I then try to use the synthesizer in Cakewalk, the GUI is replaced with a "Plugin Not Valid" message, and the console warns that Camomile can't find configuration files. I also tried the same process except renaming the Camomile.vst3 file as modules.vst3 (corresponding to modules.pd, the name of the synthesizer file), with the same erroneous result. I am unable to determine how much, if any, of the "How to create new plugins" page of the documentation applies here. It's just a bit too much technicality without context for me to understand what parts of it are required just to get going and what parts are customization options.

My first question is, are patches created using Camomile itself, or are they created independently using Pure Data, with Camomile being applied to them only after they are already created? In other words, do I need to copy the contents of the downloaded synthesizer files and paste them into new Camomile patches?

My second question is, what other steps are required to load the downloaded synthesizer files into Cakewalk using Camomile?

**Comment by pierreguillot** [comment]
> When I then try to use the synthesizer in Cakewalk, the GUI is replaced with a "Plugin Not Valid" message, and the console warns that Camomile can't find configuration files.

In the examples, there is a configuration file (MyPluginName.txt) used to define the properties of the plugin without it, Camomile doesn't know what are the parameters, the number of inputs and outputs, the type of the plugin (effect or instrument) if the plugin wants MIDI, etc. You should add it to make it work. 

For further information, you can read the section "How to create new plugins" in the wiki and the subsection https://github.com/pierreguillot/Camomile/wiki/How-to-create-new-plugins#all-plugin-properties.

> My first question is, are patches created using Camomile itself, or are they created independently using Pure Data, with Camomile being applied to them only after they are already created? In other words, do I need to copy the contents of the downloaded synthesizer files and paste them into new Camomile patches?

The patches are created with Pure Data and integrated into the plugin.

> My second question is, what other steps are required to load the downloaded synthesizer files into Cakewalk using Camomile?

I suggest you read the section "How to create new plugins" in the wiki (and also the papers, and try to understand how the examples work) and then try to create your own plugin.
